src/execution_plan/ops/op_delete.c

```diff
@@ -36,16 +42,24 @@ void _DeleteEntities(OpDelete *op) {
 		}
 	}
 
+	if(edge_count <= EDGE_BULK_DELETE_THRESHOLD) {
+		for(uint i = 0; i < edge_count; i++) {
+			Graph_DeleteEdge(g, op->deleted_edges + i);
+		}
+		edge_deleted += edge_count;
+		edge_count = 0;
+	}
+
```

Comment:
I think this optimization should be part of `Graph_BulkDelete` without modifying the `edge_count`